### ISSUE TITLE
Improve documentation interface

### ISSUE THREAD

**Issue by ManrajGrover** [ManrajGrover/algorithms-js#23]
Better interface is needed (this could be some other documentation generation tool, if required)

**Comment by gaurav-nelson** [comment]
@manrajgrover Are you still keen on this issue. I can help you by setting up documentation on GitHub pages using [Slate](https://github.com/lord/slate). algorithms-js better suits for this rather than docasauras (which is a beast to maintain). Let me know. 🥂

**Comment by manrajgrover** [comment]
@gaurav-nelson Yes, please. I'm open to any tool which serves the purpose. Are you interested in taking this one up?

**Comment by gaurav-nelson** [comment]
@manrajgrover sure, I will try to put something up after the weekend and we can take it from there.

**Comment by gaurav-nelson** [comment]
@manrajgrover To get started, please do the following:
1. Fork the Slate repo (https://github.com/lord/slate)
2. Rename the forked slate repo to `algorithms-js-docs`
3. Give contributor rights to this repo to me (so that I can easily modify content without creating PR's)

Once you will set this up, I will go ahead and update the documentation. Once done, we can review (and modify) the docs as needed and then deploy the docs to `manrajgrover/algorithms-js` in `gh-pages` branch. Later we can then setup Travis job to push updates on every merge in our docs repo.

**Comment by gaurav-nelson** [comment]
@manrajgrover Slate has a script called deploy.sh
To deploy to another repo we will add this original repo as a remote to the docs repo, then we can specify the remote in the script. The script then handles creating and pushing updates to the gh-pages branch. Someone (you) who have permissions to both the repos can run deploy.sh to publish first version and later we can setup a Travis job to handle this automatically.

**Comment by manrajgrover** [comment]
@gaurav-nelson Sounds great then. I already had a fork of slate project and have added you as a collaborator. Feel free to ping me if you face any issue
